src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -417,7 +417,8 @@ ClusterHasKnownMetadataWorkers()
 bool
 ShouldSyncTableMetadata(Oid relationId)
 {
-	if (!OidIsValid(relationId) || !IsCitusTable(relationId))
+	if (!EnableMetadataSync ||
```

Comment:
The weird side-effect of this change is that if this setting is set to OFF on the **workers**, the workers do not acquire the remote locks. In one sense, that is sort of expected, in another sense, this relation is non-trivial to consider.  Any suggestions?

```C
	bool shouldSyncMetadata = ShouldSyncTableMetadata(firstDistributedTableId);
	if (shouldSyncMetadata)
	{
		LockShardListMetadataOnWorkers(ExclusiveLock, shardList);
	}
```

(similar for `AcquireDistributedLockOnRelations`)

Reply:
We could change the check to something like `shouldSyncMetadata || !IsCoordinator()`

Reply:
> We could change the check to something like shouldSyncMetadata || !IsCoordinator()

Sounds good, so we always enforce to acquire the locks if you run from the workers. 

One related question I had was whether `LockShardListMetadataOnWorkers` can tolerate if the shard metadata is missing on the remote nodes, and yes, we gracefully handle that.